exec/java-exec/src/main/java/org/apache/drill/exec/server/options/SystemOptionManager.java

```diff
@@ -251,7 +254,12 @@ public void setOption(final OptionValue value) {
     final String name = value.name.toLowerCase();
     final OptionValidator validator = getValidator(name);
 
-    validator.validate(value); // validate the option
+    /* If the validator depends on other options */
+    if (validator instanceof DependentTypeValidators) {
+      ((DependentTypeValidators)validator).validate(value, this); // validate the option
```

Comment:
The validation is also done in the FallbackOptionManager. A testcase is present which checks the validation when changing the option in the session.